lib/engine/game/g_18_uruguay/step/farm.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/assign'
+
+module Engine
+  module Game
+    module G18Uruguay
+      module Step
+        class Farm2 < Engine::Step::Assign
+          def setup
+            @farm_id = nil
+          end
+
+          def goods
+            @game.abilities_ignore_owner(current_entity, :assign_hexes, time: 'or_start', strict_time: false)
+          end
+
+          def used_this_or?
+            goods
+          end
+
+          def blocking_for_farm?
+            return false unless @round.operating?
+
+            used_this_or?
+          end
+
+          def actions(entity)
+            return [] unless entity.minor?
+
+            %w[assign].freeze
+          end
+
+          def description
+            'Deliver cattle'
+          end
+
+          def active?
+            blocking_for_farm?
+          end
+
+          def blocks?
+            active?
+          end
+
+          def neighbor_to_choosen_farm?(farm_id, hex_id)
+            @game.hex_by_id(farm_id).neighbors.find do |neighbor|
+              neighbor[1].id == hex_id \
+              && neighbor[1].tile.city_towns.size.positive?
+            end
+          end
+
+          def available_hex(entity, hex)
+            return unless entity.minor?
+
+            return true if !@farm_id.nil? && neighbor_to_choosen_farm?(@farm_id, hex.id)
+
+            # Is it mine?
+            ability = entity.abilities.find { |a| a.type == :assign_hexes }
+            return unless ability
+            return unless ability.hexes&.include?(hex.id)
+
+            # Do we have goods?
+            return false if !@farm_id.nil? || !hex.assignments.keys.find { |a| a.include? 'GOODS' }
```

Comment:
You're using `find` but not doing anything with the value returned (other than testing it exists). This would be better as `hex.assignments.none? { |a, _| a.start_with?('GOODS') }`.

Reply:
Updated